.pre-commit-hooks.yaml

```diff
@@ -70,3 +81,121 @@
     language: python
     types:
     - file
+
+-   id: validate-repo-structure
```

Comment:
This is an example of this method of validation falling short and being uber prone to errors.

This hook will happily pass if structure is wrong, but in the checked commit there are no changes to `.ttl` files, because git has no notion of tracking changes to directory and we are using an hammer to drive a screw.

Even worse in the absolute worst case scenario where people have the most absurd directory structure you can think of with random non {ontology,vocabulary,schema} files in it, this check would still pass. It needs to have at least something right (the `.ttl` files) to even start, which is not what you want in a validator.

Reply:
the check is carried out on the ttl files (the presence of which is mandatory) and from these it verifies that the repo structure is correct. I don't understand what the problem is.

it is obvious that if there are no ttl files you can create any structure.

Reply:
I have modified it. Now it checks any file under the assets directory and the check is case-insensitive.